### PR TITLE
feat(List): move item's icon out of the title

### PR DESCRIPTION
**Pull request by mtikhinskaya**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

The list item's icon is rendered inside item's title

**What is the new behavior?**

The list item's icon is separated from its title

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/List/Items/Item/Item.component.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import React from 'react';
 import classNames from 'classnames';
 
 import ItemTitle from '../ItemTitle';
+import Icon from '../../../Icon';
 import { Actions } from '../../../Actions';
+import Column from '../Column.properties';
```
Comment (sdwvit) [inline src/List/Items/Item/Item.component.js]: nice approach

File: src/List/Items/ItemTitle/ItemTitle.component.js [hunk 2]
```diff
@@ -85,6 +84,10 @@ class TitleInput extends React.Component {
 			id={this.props.id}
 			value={this.props.value}
 			ref={(input) => { this.titleInput = input; }}
+			onChange={event => this.props.onEditChange(event, {
```
Comment (sdwvit) [inline src/List/Items/ItemTitle/ItemTitle.component.js]: maybe add debounce to prevent spamming?

Comment (mtikhinskaya) [inline src/List/Items/ItemTitle/ItemTitle.component.js]: I will create PR with ItemTitle refactor and add debounce there

File: src/List/Items/Item/Item.component.js (hunk 1 above)
Comment (mtikhinskaya) [inline src/List/Items/Item/Item.component.js]: thanks

**Pull request merged**